File: Client/lib/src/features/admin/storyline/domain/story_entry.dart
import 'package:freezed_annotation/freezed_annotation.dart';
import 'package:kalinar/src/features/admin/storyline/domain/book_page.dart';

part 'story_entry.freezed.dart';
part 'story_entry.g.dart';

@freezed
class StoryEntry with _$StoryEntry {
  const factory StoryEntry({
    required String id,
    required String title,
    required int order,
    required String type,
    String? description,
    String? date,
    String? imageUrl,
    @Default([]) List<BookPage> pages,
    @Default(false) bool isUnlocked,
  }) = _StoryEntry;

  factory StoryEntry.fromJson(Map<String, dynamic> json) => _$StoryEntryFromJson(json);
}

File: Server/Hero.Server.Core/Models/Storyline/StoryEntry.cs
namespace Hero.Server.Core.Models.Storyline
{
    public class StoryEntry
    {
        public Guid Id { get; set; }
        public Guid GroupId { get; set; }

        public string Title { get; set; }
        public string? ImageUrl { get; set; }
        public string? Description { get; set; }
        public string? Date { get; set; }

        public bool IsUnlocked { get; set; }
        public DateTime UpdatedAt { get; set; }
        public int Order { get; set; }

        public virtual void Update(StoryEntry updated)
        {
            this.Title = updated.Title;
            this.ImageUrl = updated.ImageUrl;
            this.Description = updated.Description;
            this.IsUnlocked = updated.IsUnlocked;
            this.Date = updated.Date;
            this.UpdatedAt = DateTime.UtcNow;
        }
    }
}
